sw.js

```diff
@@ -1,74 +1,45 @@
-self.global = self // eslint-disable-line
+/* eslint-env browser, serviceworker */
 
-require('debug').enable('planktos:*')
-const debug = require('debug')('planktos:sw')
 const planktos = require('.')
-const injection = require('./lib/injection')
 
-const scope = global.location.pathname.substring(0, global.location.pathname.lastIndexOf('/'))
+// The location of the planktos root directory
+const root = location.pathname.substring(0, location.pathname.lastIndexOf('/'))
 let available = {}
 let delegator = null
 
-global.addEventListener('fetch', onFetch)
-global.addEventListener('activate', onActivate)
-global.addEventListener('install', onInstall)
-global.addEventListener('message', onMessage)
+addEventListener('fetch', onFetch)
+addEventListener('install', onInstall)
+addEventListener('message', onMessage)
 
 assignDelegator()
 
 function onFetch (event) {
   let url = new URL(event.request.url)
-  let fpath = planktos._normalizePath(url.pathname.replace(scope, ''))
-  let search = url.search.substr(1).split('&')
 
-  if (url.host !== global.location.host || event.request.method !== 'GET') return
-  if (planktos.preCached.indexOf('/' + fpath) === -1 && fpath.startsWith('planktos/')) return
+  if (url.host !== location.host || event.request.method !== 'GET') return
+  if (url.pathname.replace(root, '').startsWith('/planktos/files/')) return
```

Comment:
This occurs when the request is a webseed . A comment explaining this would be helpful.